bundles/org.openhab.binding.modbus.helioseasycontrols/src/test/java/org/openhab/binding/modbus/helioseasycontrols/internal/PreparePayloadTest.java

```diff
@@ -0,0 +1,60 @@
+package org.openhab.binding.modbus.helioseasycontrols.internal;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.MethodSource;
+import org.openhab.io.transport.modbus.ModbusRegisterArray;
+
+/**
+ * @author Sami Salonen - Initial contribution
+ */
+public class PreparePayloadTest {
+
+    private Method preparePayloadMethod;
+
+    public PreparePayloadTest() throws NoSuchMethodException, SecurityException {
+        preparePayloadMethod = HeliosEasyControlsHandler.class.getDeclaredMethod("preparePayload", String.class);
+        preparePayloadMethod.setAccessible(true);
+    }
+
+    private ModbusRegisterArray preparePayload(String payload) {
+        try {
+            return (ModbusRegisterArray) preparePayloadMethod.invoke(null, payload);
+        } catch (IllegalAccessException | IllegalArgumentException | InvocationTargetException e) {
+            fail("Reflection failure:" + e.getMessage());
+            throw new RuntimeException(); // to make compiler happy
+        }
+    }
+
+    public static Collection<Object[]> data() {
+        return Collections.unmodifiableList(Stream
+                // Due to nul byte, full register full of nul bytes added
+                .of(new Object[] { "v00020=a", new ModbusRegisterArray(0x7630, 0x3030, 0x3230, 0x3d61, 0x0000) },
+                        new Object[] { "v00020=aa", new ModbusRegisterArray(0x7630, 0x3030, 0x3230, 0x3d61, 0x6100) })
+                .collect(Collectors.toList()));
+    }
+
+    @ParameterizedTest
+    @MethodSource("data")
+    public void testPreparePayload(String payload, ModbusRegisterArray expectedRegisters)
+            throws IllegalAccessException, IllegalArgumentException, InvocationTargetException {
```

Comment:
are these throws necessary?

Reply:
These are from the reflection, so I believe they are? Alternatively I could try catch and fail/throw unchecked exception

Reply:
The checked exceptions `IllegalAccessException` and `InvocationTargetException` are never thrown. So they could be removed. Specifying an unchecked exception like `IllegalArgumentException` has no effect in tests, AFAIK. So, that could be removed, too.

Reply:
Resolved in e43bdc610